libstuff/STCPNode.cpp

```diff
@@ -341,6 +388,9 @@ void STCPNode::Peer::reset() {
     latency = 0;
     loggedIn = false;
     priority = 0;
+    if (socket) {
+        delete socket;
+    }
```

Comment:
If we had a unique_ptr to this Socket, we wouldn't need this block (maybe in the next step of this refactor)